[CMake] Use imported targets in find module for WPEBackend-fdo

aperezdc opened the pull request:
#### 373e7435c3ccae2153b19a41f3c09f4bd4bcc8ef
<pre>
[CMake] Use imported targets in find module for WPEBackend-fdo
<a href="https://bugs.webkit.org/show_bug.cgi?id=270167">https://bugs.webkit.org/show_bug.cgi?id=270167</a>

Reviewed by Michael Catanzaro.

Rename WPEBackend-fdo find module to WPEBackendFDO, make it define a
WPE::FDO imported target, and change the rest of the CMake build system
to use it. Simplify along the way and sprinkle a few changes from
&lt;target&gt;_LIBRARIES to &lt;target&gt;_PRIVATE_LIBRARIES.

* Source/WebCore/PlatformWPE.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/cmake/FindWPEBackendFDO.cmake: Added.
* Source/cmake/FindWPEBackend_fdo.cmake: Removed.
* Source/cmake/GStreamerChecks.cmake:
* Source/cmake/OptionsWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/TestWebKitAPI/PlatformWPE.cmake:
* Tools/TestWebKitAPI/glib/PlatformWPE.cmake:
* Tools/WebKitTestRunner/PlatformWPE.cmake:
* Tools/wpe/backends/PlatformWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/275429@main">https://commits.webkit.org/275429@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c305fc49dbb520c1c34a7aefb868c3483e8b1a90

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41722 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20736 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44103 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44297 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37815 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44029 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23872 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18067 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34481 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42296 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17663 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35930 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15152 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15355 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36952 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45684 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/35175 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37907 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37273 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41023 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/41348 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16537 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13576 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39455 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18156 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/48359 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18213 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9851 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5603 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17800 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
The attempt to get the version number from the header files seems like overkill to me -- that's just there in case pkg-config doesn't work? -- but whatever. More imported targets is good.

aperezdc committed:
[CMake] Use imported targets in find module for WPEBackend-fdo
https://bugs.webkit.org/show_bug.cgi?id=270167

Reviewed by Michael Catanzaro.

Rename WPEBackend-fdo find module to WPEBackendFDO, make it define a
WPE::FDO imported target, and change the rest of the CMake build system
to use it. Simplify along the way and sprinkle a few changes from
<target>_LIBRARIES to <target>_PRIVATE_LIBRARIES.

* Source/WebCore/PlatformWPE.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/cmake/FindWPEBackendFDO.cmake: Added.
* Source/cmake/FindWPEBackend_fdo.cmake: Removed.
* Source/cmake/GStreamerChecks.cmake:
* Source/cmake/OptionsWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/TestWebKitAPI/PlatformWPE.cmake:
* Tools/TestWebKitAPI/glib/PlatformWPE.cmake:
* Tools/WebKitTestRunner/PlatformWPE.cmake:
* Tools/wpe/backends/PlatformWPE.cmake:

Canonical link: https://commits.webkit.org/275429@main
